### PR TITLE
eliminate callbacks for objectives/competencies in single event component

### PR DESCRIPTION
**Pull request by stopfstedt**
companion ticket to ilios/ilios#2173

### CODE REVIEW

**Comment by jrjohnson** [comment]
Deploy preview for *ilios-common* ready!

Built with commit 9bde0f006e480a9dea85af7310257b91d3d3fd33

https://deploy-preview-246--ilios-common.netlify.com

File: addon/components/single-event.js [hunk 1]
```diff
@@ -168,4 +177,34 @@ export default Component.extend({
...
+  /**
+   * Callback function for <code>Array.sort()<code>.
+   * Compares two given Objects by their position property (in ascending order), and then by id (descending).
+   *
+   * @method positionSortingCallback
+   * @param {Object} obj1
+   * @param {Object} obj2
+   * @return {Number}
+   */
+  positionSortingCallback(obj1, obj2) {
```
Comment (jrjohnson) [inline addon/components/single-event.js]: Why not use the `sortable-by-position` mixin here?

File: addon/components/single-event.js [hunk 2]
```diff
@@ -99,31 +102,37 @@ export default Component.extend({
...
+      }
       return {
+        id: objective.id,
         title,
-        domain: competency.get('title') + ' (' + domain.get('title') + ')',
+        domain: competency.title + ' (' + domain.title + ')',
         position
       };
-    });
+    }).sort(this.positionSortingCallback);
```
Comment (jrjohnson) [inline addon/components/single-event.js]: Just curious - how were we sorting this before? I don't see any where that was happening.

File: addon/components/single-event.js (hunk 1 above)
Comment (stopfstedt) [reply addon/components/single-event.js]: `sortable-by-position` mixin assumes Ember objects as input. we're operating on plain on javascript objects here, so `get()` won't work here. so i re-implemented that sorting callback to work with POJsOs.

File: addon/components/single-event.js (hunk 2 above)
Comment (stopfstedt) [reply addon/components/single-event.js]: we're using the output of the `sortedObjectives` CP before, which uses the callback provided by  `sortable-by-position` mixin.

File: addon/components/single-event.js (hunk 1 above)
Comment (jrjohnson) [reply addon/components/single-event.js]: Ah ok. Thanks.

**Pull request merged**